### PR TITLE
feat(checkbox): enhance CheckboxSelectors

### PR DESCRIPTION
**Pull request by aelfaleh**
### Proposed Changes

Enhance CheckboxSelectors to return GroupableCheckboxesState
https://coveo.slack.com/archives/C01M10DME91/p1645467040489239

### Acceptance Criteria

-   [x] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/react/src/components/checkbox/CheckboxSelectors.ts [hunk 1]
```diff
@@ -1,10 +1,29 @@
...
+        });
+        if (groupableCheckboxesState) {
+            const checkboxState: ICheckboxState = groupableCheckboxProps.isParent
+                ? groupableCheckboxesState.parent
+                : _.findWhere(groupableCheckboxesState.checkboxes, {id: id});
+            return checkboxState;
+        }
+    }
+    return _.findWhere(state.checkboxes, {id});
+};
```
Comment (toofff) [inline packages/react/src/components/checkbox/CheckboxSelectors.ts:26]:
```suggestion
const get = (
    state: PlasmaState,
    {id}: {id: string},
    isGroupableState?: boolean,
    groupableCheckboxProps?: IGroupableCheckboxOwnProps
): ICheckboxState | undefined => {
    if (isGroupableState) {
        const groupableCheckboxesState: IGroupableCheckboxesState = _.findWhere(state.groupableCheckboxes, {
            parentId: groupableCheckboxProps.parentId || groupableCheckboxProps.id,
        });
        if (groupableCheckboxesState) {
            return groupableCheckboxProps.isParent
                ? groupableCheckboxesState.parent
                : _.findWhere(groupableCheckboxesState.checkboxes, {id: id});
        }
    }
    return _.findWhere(state.checkboxes, {id});
};

```

**Review by nathanlb — COMMENTED** [review]
This will probably work but I'm wondering if it would be better to create a separate set of selectors for groupable checkboxes given it has it's own separate state and this adds complexity to an otherwise very streamlined selector for regular checkboxes.

**Review by nathanlb — APPROVED** [review]

File: packages/react/src/components/checkbox/CheckboxSelectors.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (toofff) [reply packages/react/src/components/checkbox/CheckboxSelectors.ts]: This little change is very good, removing the indentation from the code by reversing the condition and moving the last return to the start of the method. it makes the code easier to read :+1:

**Review by toofff — APPROVED** [review]
OK for me after the correction of the last fix proposed by @nathanlb

**Review by GermainBergeron — APPROVED** [review]

**Pull request merged**